usersync/usersyncers/syncer_test.go

```diff
@@ -7,40 +7,30 @@ import (
 	"github.com/prebid/prebid-server/openrtb_ext"
 )
 
-func TestSyncers(t *testing.T) {
-	cfg := &config.Configuration{}
-	syncers := NewSyncerMap(cfg)
-	for _, bidderName := range openrtb_ext.BidderMap {
-		if _, ok := syncers[bidderName]; !ok {
-			t.Errorf("No syncer exists for adapter: %s", bidderName)
-		}
+func TestNewSyncerMap(t *testing.T) {
+	cfg := &config.Configuration{
+		Adapters: map[string]config.Adapter{
```

Comment:
In an effort to isolate adapters entirely I was thinking of doing something like this: https://github.com/zachbadgett/prebid-server/commit/57a580ca4fd46a3d1cb07e3d3c2c52c383a0a3ba#diff-25c7c8df3a616bd75098f16d158d0a0bR29

This could still achieve your goal of "impossible" to do wrong by doing a panic on `Register`.

The `usersync/usersyncers/` would ultimately be removed.

Reply:
Keep in mind the design decisions I made in that commit attempts to change other code as little as possible (especially in the adapters themselves).

Reply:
yeah, that's fair.

I'm trying to make sure that the code is left in a good state after each PR. Someone might try to submit a Bidder between now and the time the next PR merges, or the next PR might never come.

We might also spend a while discussing tradeoffs. For `init` & `Register` in particular, I know we'll have at least some back and forth. I'm most definitely not an experienced Gopher... but I can think of some good language-agnostic reasons to avoid that structure, and the benefits that I can think of don't outweigh the costs. As an avid Go supporter, I'm curious to learn about how you weigh them :).

That's a long-winded way of saying: if the future changes don't pan out, I'd rather not lose the value provided by this test.